docs/udf.rst

```diff
@@ -0,0 +1,367 @@
+          }
+        };
+      }
+
+    }
+
+UdafDescription Annotation
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+The ``@UdafDescription`` annotation is applied at the class level and has four fields, two of which are required:
+
++------------+------------------------------+---------+
+| Field      | Description                  | Required|
++============+==============================+=========+
+| name       | The name of the function(s)  | Yes     |
+|            | represented by this class.   |         |
++------------+------------------------------+---------+
+| description| A string describing generally| Yes     |
+|            | what the function(s) in this |         |
+|            | class do.                    |         |
++------------+------------------------------+---------+
+| author     | The author of the UDF.       | No      |
++------------+------------------------------+---------+
+| version    | The version of the UDF.      | No      |
++------------+------------------------------+---------+
+
+
+UdafFactory Annotation
+~~~~~~~~~~~~~~~~~~~~~~
+
+The ``@UdafFactory`` annotation is applied to public static methods of a class annotated with ``@UdafDescription``.
+The method must return either ``Udaf``, or, if it supports table aggregations, ``TableUdaf``.
+Each annotated method is a factory for an invocable aggregate function in KSQL. The annotation only has a single
+field ``description`` that is required. You can use this to better describe what a particular version
+of the UDF does, for example:
+
+.. code:: java
+
+    @UdafFactory(description = "Sums BIGINT columns.")
+    public static TableUdaf<Long, Long> createSumLong(){...}
+
+    @UdafFactory(description = "Sums the length of VARCHAR columns".)
+    public static Udaf<String, Long> createSumLengthString(final String initialString)
+
+
+===============
+Supported Types
+===============
+
+The types supported by UDFs are currently limited to:
+
++--------------+------------------+
+|  Java Type   | KSQL Type        |
++==============+==================+
+| int          | INTEGER          |
++--------------+------------------+
+| Integer      | INTEGER          |
++--------------+------------------+
+| boolean      | BOOLEAN          |
++--------------+------------------+
+| Boolean      | BOOLEAN          |
++--------------+------------------+
+| long         | BIGINT           |
++--------------+------------------+
+| Long         | BIGINT           |
++--------------+------------------+
+| double       | DOUBLE           |
++--------------+------------------+
+| Double       | DOUBLE           |
++--------------+------------------+
+| String       | VARCHAR          |
++--------------+------------------+
+| List         | ARRAY            |
++--------------+------------------+
+| Map          | MAP              |
++--------------+------------------+
+
+Note: Complex types other than List and Map are not currently supported
+
+
+=========
+Deploying
```

Comment:
Please describe failure conditions such as:

- Your UDF was deployed to only a subset of KSQL servers.  What happens on those servers where the UDF is not available / available yet?  How can the user spot this error?  How are the KSQL statements affected that are using this UDF?  Will only the subset of KSQL servers process the statement that do have the UDF available locally?  Or will no server process them, or ...?

Reply:
This was addressed in a subsequent commit by @dguy .